pkg/volume/flexvolume/probe.go

```diff
@@ -38,12 +40,36 @@ func ProbeVolumePlugins(pluginDir string) []volume.VolumePlugin {
 		// then, executable will be pluginDir/dirname/cifs
 		if f.IsDir() {
 			execPath := path.Join(pluginDir, f.Name())
-			plugins = append(plugins, &flexVolumePlugin{
-				driverName:          utilstrings.UnescapePluginName(f.Name()),
-				execPath:            execPath,
-				runner:              exec.New(),
-				unsupportedCommands: []string{},
-			})
+
+			attachablePlugin := &flexVolumeAttachablePlugin{
```

Comment:
nit: Factoring is a little strange here. Two suggestions:

1. How about introducing a constructor method `NewFlexVolumePlugin(...)` that takes all the `flexVolumePlugin` fields as a parameter. It would internally test for attachability and return either a `flexVolumeAttachablePlugin` or a `flexVolumePlugin`--that'll simplify the code here.
2. For testing attachability, instead of constructing a `flexVolumePlugin` object and calling `isAttachable()` on it, it would be cleaner to have `IsAttachable(...)` be a static method that takes all the `flexVolumePlugin` fields as a parameter, make the appropriate calls to the driver to determine attachability, and then return true/false.